### ISSUE TITLE
Bug: Pydantic @computed_field not showing in OpenAPI json schema

### ISSUE THREAD

**Issue by ssenior45** [litestar-org/litestar#2792]
### Description

When using a Pydantic model with a computed field, the responses from the API calls include the computed field. 
 However, the computed field is not being added to the OpenAPI json schema.

### MCVE

```python
import json

from litestar import Litestar, get
from litestar.testing import TestClient
from pydantic import BaseModel, computed_field


class Sample(BaseModel):
    property_one: str

    @computed_field()
    def property_two(self) -> bool:
        return True


@get("/")
async def test_handler() -> Sample:
    return Sample(property_one="test")


app = Litestar(route_handlers=[test_handler], debug=True)


with TestClient(app=app) as client:
    get_response = client.get(
        "/",
    )
    print(json.dumps(get_response.json(), indent=2))

    response = client.get(
        "/schema/openapi.json",
    )
    print(json.dumps(response.json()["components"]["schemas"], indent=2))
```

### Steps to reproduce

```
1. Execute the MCVE
```

### Screenshots

None.

### Logs

When using the model with a computed field, if I check the content of the get_response:

```
{
  "property_one": "test",
  "property_two": true
}
INFO - 2023-11-28 15:06:55,933 - httpx - _client - HTTP Request: GET http://testserver.local/ "HTTP/1.1 200 OK"
```

The computed field is in the response, as expected.

However, the computed field is not being added to the json schema:

```
{
  "Sample": {
    "properties": {
      "property_one": {
        "type": "string"
      }
    },
    "type": "object",
    "required": [
      "property_one"
    ],
    "title": "Sample"
  }
}
INFO - 2023-11-28 15:06:55,938 - httpx - _client - HTTP Request: GET http://testserver.local/schema/openapi.json "HTTP/1.1 200 OK"
```

### Litestar Version

2.4.0.

### Platform

- [X] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh Litestar dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/2792">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/2792/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/2792/pledge.svg">
</picture>
</a>

**Comment by marcuslimdw** [comment]
I believe this isn't a Litestar issue, but a Pydantic one:

```
>>> Sample.model_json_schema()
{'properties': {'property_one': {'title': 'Property One', 'type': 'string'}}, 'required': ['property_one'], 'title': 'Sample', 'type': 'object'}
```

The computed property doesn't appear in the spec because Litestar just delegates to Pydantic and Pydantic doesn't generate it.

**Comment by cofin** [comment]
Resolved by #2797

**Comment by peterschutt** [comment]
Re-opening as that PR hasn't been merged yet.
